src/FetchWorkerTool.js

```diff
@@ -52,6 +54,8 @@ class PrivateFetchWorkerTool {
                         }
                     }
                 });
+
+                this.worker = worker;
```

Comment:
I like that this isolates the worker from the owning object until the worker has been fully constructed.